docs/extend/extend-cli/cli-developing-a-plugin.md

```diff
@@ -2,218 +2,339 @@
+Before you test your new command, confirm that you are able to connect to the mainframe.
+
+In order for the client-side API to reach the mainframe (to fetch data sets), Zowe CLI needs a z/OSMF profile for access. See [Using profiles](https://docs.zowe.org/stable/user-guide/cli-using-using-profiles/) for information. 
+
+Once the connection between Zowe CLI and z/OSMF is confirmed, build and install the plug-in before running it for the first time.
+
+Follow these steps:
+
+1. Repeat the steps in [Building your plug-in source](../extend-cli/cli-developing-a-plugin#building-your-plug-in-source).
+
+2. Issue the `zowe plugins install` command to install Files Util Plug-in into Zowe CLI.
+
+    A success message displays if installed correctly.
+
+3. Issue the following command (the Files Util Plug-in command) replacing the data set names with valid mainframe data set names on your system:
 
-Issue the following command. Replace the data set names with valid mainframe data set names on your system:
+    ```
+    zowe files-util diff data-sets "IBMUSER.REXX(TESTADD)" "IBMUSER.REXX(TESTSUB)"
+    ```
+    The raw diff output displays as a command response:
 
-![Pipe Output](../../images/guides/CLI/pipeOutput.png)
+    ```
+    $ zowe files-util diff data-sets "IBMUSER.REXX(TESTADD)" "IBMUSER.REXX(TESTSUB)"
+    ===================================================================
+    --- IBMUSER.REXX(TESTADD)        Old
+    +++ IBMUSER.REXX(TESTSUB)        New
+    @@ -1,6 +1,6 @@
+    /* REXX */
+    -say add(3,2)
+    +say sub(3,2)
+    exit
+    -add:
+    +sub:
+    parse arg a,b
+    -return a + b
+    +return a - b
+    ```
 
-The raw diff output is displayed as a command response:
+    When successful, the output displays plain text diffs of the entered data sets.
 
-![Raw Diff Output](../../images/guides/CLI/diffOutput.png)
+**To create a unique plug-in:** Use Step 3 to run your new command.
 
 ## Bringing together new tools!
-The advantage of Zowe CLI and of the CLI approach in mainframe development is that it allows for combining different developer tools for new and interesting uses.
 
-[diff2html](https://diff2html.xyz/) is a free tool to generate HTML side-by-side diffs to help see actual differences in diff output.
+You have created a simple CLI plug-in that provides plain text diffs of two data sets. But you may not want to end there.
+
+Depending on the complexity of your changes, it can be difficult to identify data set differences with plain text.
+
+To help this, you can extend Files Util Plug-in to create a more visual output. For this tutorial, use [diff2html](https://diff2html.xyz/) to generate side-by-side diffs that make it easier to compare changes, as seen in the image below.
+
+| ![Side by Side Diff](../../images/guides/CLI/htmlDiff2.png) |
+|:--:|
+| <b>Diff to HTML by [rtfpessoa](https://github.com/rtfpessoa)</b>|
+
+Follow these steps:
+
+1. Run `npm install -g diff2html-cli` to install `diff2html`.
```

Comment:
Should we warn the developer of what this command will do and the possible consequences?
Or at least send them to NPM's doc on what `npm install --global` does?
- https://docs.npmjs.com/cli/v8/commands/npm-install#global

Reply:
added a note.